risc0/circuit/rv32im/src/execute/executor.rs

```diff
@@ -208,6 +215,7 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
             ring: AllocRingBuffer::new(10),
             povw_job_id,
             circuit_version,
+            segment_counter: 0,
```

Comment:
BTW, this is unneeded and undefined in m3.

Reply:
Good to know. That does make me realize we'll need a new way of assigning nonces for PoVW in M3. I'm going to leave this in this PR, since its still required while v2 is supported and the change on this line is to do with extracting it onto the `Executor` struct to allow for the `split_segment` function to be factored out.